docker/python/Dockerfile

```diff
@@ -24,21 +22,25 @@ RUN rm -fr /opt/cmake/bin/cpack
 RUN ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake
 RUN ln -s /opt/cmake/bin/ctest /usr/local/bin/ctest
 
-RUN python3.7 -m pip install numpy pandas
+RUN python3.7 -m pip install numpy pandas cython codecov nose2 mock flake8 pytest pytest-cov traitlets ipywidgets faker psutil zerorpc
 
-RUN ls -al /usr/local/lib/python3.7/site-packages/numpy
-
-RUN wget https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz >/dev/null 2>&1  || echo "wget failed"
+# install boost
+RUN wget https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.gz >/dev/null 2>&1  || echo "wget boost failed"
 RUN tar xfz boost_1_71_0.tar.gz
 # https://github.com/boostorg/build/issues/468
-RUN cd boost_1_71_0 && CPLUS_INCLUDE_PATH="/usr/local/include/python3.7m" C_INCLUDE_PATH="$C_INCLUDE_PATH:/usr/local/include/python3.7m" ./bootstrap.sh --with-python=/usr/local/bin/python3.7 || echo "boostrap failed"
-RUN cd boost_1_71_0 && CPLUS_INCLUDE_PATH="$CPLUS_INCLUDE_PATH:/usr/local/include/python3.7m" C_INCLUDE_PATH="$C_INCLUDE_PATH:/usr/local/include/python3.7m" ./b2 -j4 install || echo "build boost failed" 
+RUN cd boost_1_71_0 && ./bootstrap.sh || echo "boostrap failed"
+RUN cd boost_1_71_0 && ./b2 -j4 install || echo "build boost failed" 
+
+RUN python3 -m pip install 'numpy>=1.13.1' 'pandas>=0.22.0'
 
-RUN ln -s /usr/local/lib/libboost_python37.so /usr/local/lib/libboost_python.so
-RUN ln -s /usr/local/lib/libboost_numpy37.so /usr/local/lib/libboost_numpy.so
+# install pyarrow
+RUN wget https://github.com/apache/arrow/archive/apache-arrow-0.15.0.tar.gz >/dev/null 2>&1  || echo "wget arrow failed"
+RUN tar xfz apache-arrow-0.15.0.tar.gz
+RUN cd arrow-apache-arrow-0.15.0 && mkdir build && cd build && cmake ../cpp/ -DARROW_RPATH_ORIGIN=ON -DARROW_PYTHON=ON -DARROW_FLIGHT=ON -DARROW_COMPUTE=ON -DCMAKE_INSTALL_PREFIX=/usr -DARROW_DATASET=OFF -DARROW_BUILD_UTILITIES=OFF -DARROW_JEMALLOC=OFF -DARROW_DEPENDENCY_SOURCE=BUNDLED -DARROW_HDFS=OFF -DARROW_WITH_BACKTRACE=OFF -DARROW_WITH_BROTLI=OFF -DARROW_WITH_BZ2=OFF -DARROW_WITH_LZ4=OFF -DARROW_WITH_SNAPPY=OFF -DARROW_WITH_ZLIB=OFF -DARROW_WITH_ZSTD=OFF && make -j2 && sudo make install && cd ../python && sudo python3.7 setup.py install
 
-RUN python3 -m pip install codecov nose2 mock flake8 pytest pytest-cov traitlets ipywidgets faker psutil zerorpc
-RUN python3 -m pip install 'numpy>=1.13.1' 'pandas>=0.22.0' 'pyarrow==0.15.0'
```

Comment:
Shouldn't we be testing the way users are wont to have pyarrow installed, namely `pip install pyarrow`, rather than build from source here?

Reply:
Perspective is not compatible with the manylinux pyarrow wheel, if you can get perspective to build on centos6 with gcc 4.8 (a non C++11 compiler) then we can use the pyarrow wheel https://www.python.org/dev/peps/pep-0571/#id48

Reply:
I propose if this is something we want to maintain tests for on travis, we should build a separate custom container for it, and either way we want to maintain this container as-is for python3 testing.  E.g. this should be reverted.